File: thesis/src/scripts/mathjax.lua
---
--- Filter MathJax double dollars $$ to single dollars for use in LaTeX and friends.
---

local f = io.open(filename, "r")
local contents = f:read("*all")
f:close()
local filtered = contents:gsub("%$%$", "$")
local name = filename:gsub("(.*/)(.*)", "%2")
local r = io.open(name .. ".filtered", "w")
r:write(filtered, "\n")
r:close()
return name .. ".filtered"

File: thesis/scripts/mathjax.py
"""
Script that will remove mathjax stuff from markdown files.
"""
import argparse
import os

parser = argparse.ArgumentParser(description='Process Markdown for Mathjax.')
parser.add_argument('input', type=str, help='filename for input')
parser.add_argument('output', type=str, help='filename for output')

args = parser.parse_args()

with open(args.input, "r") as f_in:
    contents = f_in.read()

filtered = contents.replace("$$", "$")

os.makedirs(os.path.dirname(args.output), exist_ok=True)

with open(args.output, "w") as f_out:
    f_out.write(filtered)
